Clean up code

darrenscerri opened the pull request:
React’s `render` function never passes arguments

Commit:
render never takes arguments

React’s `render` function never passes arguments

Commit:
Clean up killEvent

goncalvesjoao commented on src/injectProps.js:
```diff
@@ -1,8 +1,8 @@
 
-  descriptor.value = function propsInjectorFunction(...args) {
-    return oldFunction.bind(this)(...args, this.props);
+  descriptor.value = function propsInjectorFunction() {
```
This prevents other people from using the decorator on other methods, like it is describe on the documentation: https://croudcare.github.io/relpers/ 

Maybe we should create another decorator called injectPropsWithArgs?

darrenscerri replied:
In that case I would suggest that the props are passed always as the first argument as otherwise the argument index would vary depending on the number of arguments which the function was invoked with.

goncalvesjoao commented on src/killEvent.js:
```diff
@@ -1,17 +1,14 @@
+    event.preventDefault();
 
-    oldHandler.apply(this, args);
+    return target.apply(this, arguments);
```
I agree.
Just for the record, we are following the airbnb style guide: https://github.com/airbnb/javascript#7.6 that advises against using "arguments".
Your changes do not go against what airbnb was talking about and I get that. Thanks

darrenscerri replied:
Good point. We're not doing anything in particular with arguments rather than passing them to the wrapped function so I guess that's acceptable.

Commit:
Always pass props as first argument

Commit:
Merge remote-tracking branch 'croudcare/master'

goncalvesjoao commented on src/killEvent.js:
```diff
@@ -1,16 +1,12 @@
+    return target.apply(this, arguments);
   }
 
-  if (typeof target === 'function') { return eventCanceler; }
```
Sorry @darrenscerri the documentation on killEvent was not complete, I've updated it: https://croudcare.github.io/relpers/v1.1/ the killEvent was originally meant to be used has a normal function, removing this will break some people's code.

The v1.1 version as specs covering 100% of the code, check them out to see it your changes break anything.

goncalvesjoao commented on src/injectProps.js:
```diff
@@ -2,7 +2,7 @@ function injectProps(target, name, descriptor) {
 
   descriptor.value = function propsInjectorFunction(...args) {
-    return oldFunction.bind(this)(...args, this.props);
+    return oldFunction.bind(this)(this.props, ...args);
```
> In that case I would suggest that the props are passed always as the first argument as otherwise the argument index would vary depending on the number of arguments which the function was invoked with.

you are talking about optional parameters right? This will bump the version to v1.2 but I agree.

goncalvesjoao committed:
travis setup

goncalvesjoao committed:
travis setup update

Commit:
Merge remote-tracking branch 'croudcare/master'

Commit:
Allow killEvent to be used directly on a function

goncalvesjoao commented:
@darrenscerri try to merge master again, travis wasn't properly configured at the time you've pulled the .travis.yml to your repo

Commit:
Merge remote-tracking branch 'croudcare/master'

Conflicts:
.travis.yml

Commit:
Fix killEvent to enable direct usage

Commit:
Fix tests for injectProps

Fix tests for injectProps for when function accepts arguments. Props
should always be passed as the first argument

Commit:
Bump to v1.2.0

goncalvesjoao commented:
thanks for all the trouble @darrenscerri
